### PR TITLE
DEPR: remove PanelGroupBy, disable DataFrame.to_panel

### PR DESCRIPTION
**Pull request by jbrockmendel**
My understanding is that we're removing Panel in 0.25.0.  A local attempt to do this all-at-once got messy quick (largely due to io.pytables and io.msgpack).  This gets the ball rolling by removing only PanelGroupBy and DataFrame.to_panel, followed by all of the code+tests that rely on either of these.

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
looks good. I assumed we will need multiple passes to completely remove Panel.
- can you add a whatsnew note in Deprecation Removals (can then add each issue number to it as they are done). Maybe want to create an issue to track all of the things to be removed, e.g. the places where added NotIMplementedErorr and asvs.

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -3944,29 +3944,7 @@ def read(self, where=None, columns=None, **kwargs):
                 objs.append(obj)
 
         else:
-            warnings.warn(duplicate_doc, DuplicateWarning, stacklevel=5)
```
Comment (jbrockmendel) [inline pandas/io/pytables.py]: @jreback this is the part of this PR (and the larger process) that I'm least confident about.  Are we sure we want to rip this out entirely or should it somehow be adapted to not use Panel?

**Comment by jbrockmendel** [comment]
> can you add a whatsnew note in Deprecation Removals

Done.

**Comment by jbrockmendel** [comment]
Travis fail fixed

**Comment by jbrockmendel** [comment]
Worth pursuing or close?

File: pandas/tests/groupby/test_grouping.py [hunk 2]
```diff
@@ -564,52 +563,6 @@ def test_list_grouper_with_nat(self):
...
-        grouped = wp.groupby(lambda x: x.month, axis='major')
-
-        gp = grouped.get_group(1)
-        expected = wp.reindex(
-            major=[x for x in wp.major_axis if x.month == 1])
-        assert_panel_equal(gp, expected)
-
-        # GH 5267
-        # be datelike friendly
-        df = DataFrame({'DATE': pd.to_datetime(
```
Comment (WillAyd) [inline pandas/tests/groupby/test_grouping.py]: Hmm should we keep this DataFrame test around?

File: pandas/io/pytables.py [hunk 3]
```diff
@@ -4875,16 +4853,3 @@ def select_coords(self):
             return self.coordinates
 
         return np.arange(start, stop)
-
-# utilities ###
-
-
-def timeit(key, df, fn=None, remove=True, **kwargs):
```
Comment (WillAyd) [inline pandas/io/pytables.py]: Just to confirm this is dead code and not necessarily related to Panel right?

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/io/pytables.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/io/pytables.py]: Yes.  Good idea double-checking.

File: pandas/tests/groupby/test_grouping.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/groupby/test_grouping.py]: Good catch.  It isn't always obvious when a multi-part test has pieces that can be salvaged.  Will update.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jbrockmendel 

keep em coming on this! (which should do before we remove PY2 stuff).

**Comment by jorisvandenbossche** [comment]
> My understanding is that we're removing Panel in 0.25.0.

I think what we discussed this summer was to have all bigger deprecations in 0.25 and remove them only before 1.0 (this is what is written down in https://github.com/pandas-dev/pandas/wiki/Pandas-Sprint-(July,-2018), and Panel is also listed in the section to remove in 1.0 in https://github.com/pandas-dev/pandas/issues/6581).

But, given that Panel is already deprecated for a long time (and the process has already started now ..), I don't think I mind that much removing it already. But just wanted to note that this might warranted some more discussion before directly starting to remove things.

**Pull request merged**